sdk/Sdk.Analyzers/BindingTypeNotSupported.cs

```diff
@@ -0,0 +1,96 @@
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.Azure.Functions.Worker.Sdk.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class BindingTypeNotSupported : DiagnosticAnalyzer
+    {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(DiagnosticDescriptors.BindingTypeNotSupported); } }
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze);
+            context.RegisterSymbolAction(AnalyzeMethod, SymbolKind.Method);
+        }
+
+        private static void AnalyzeMethod(SymbolAnalysisContext context)
+        {
+            var method = (IMethodSymbol)context.Symbol;
+
+            if (!method.IsFunction(context))
+            {
+                return;
+            }
+
+            var methodParameters = method.Parameters;
+
+            if (method.Parameters.Length == 0)
+            {
+                return;
+            }
+
+            foreach (var parameter in methodParameters)
+            {
+                foreach (var attribute in parameter.GetAttributes())
+                {
+                    var attributeType = attribute?.AttributeClass;
+                    if (!attributeType.IsInputOrTriggerBinding())
+                    {
+                        continue;
+                    }
+
+                    var inputConverterAttributeType = context.Compilation.GetTypeByMetadataName(Constants.Types.InputConverterAttribute);
+                    var inputConverterAttributes = attributeType.GetAttributes().Where(attr => attr.AttributeClass.Equals(inputConverterAttributeType)).ToList();
+
+                    if (inputConverterAttributes.Count == 0)
+                    {
+                        continue;
+                    }
+
+                    var supportedTypes = new List<object>();
+
+                    foreach (var inputConverterAttribute in inputConverterAttributes)
+                    {
+                        var converterName = inputConverterAttribute.ConstructorArguments.FirstOrDefault().Value.ToString();
+                        var converter = context.Compilation.GetTypeByMetadataName(converterName);
+
+                        var converterAttributes = converter.GetAttributes();
+
+                        supportedTypes.AddRange(converterAttributes
+                                                    .Where(a => a.AttributeClass.Name == Constants.Names.SupportedConverterTypeAttribute)
+                                                    .SelectMany(a => a.ConstructorArguments.Select(arg => arg.Value))
+                                                    .ToList());
+
+                        var converterSupportsJsonDeserialization = converterAttributes.Any(a => a.AttributeClass.Name == Constants.Names.SupportsJsonDeserializationAttribute);
```

Comment:
Unsure of how to also check for IEnumerable as well? Just because a converter supports JsonDeserialization, it doesn't mean it supports lists?